hack/lib/catalogsource.bash

```diff
@@ -214,3 +215,23 @@ function add_user {
 
   logger.info "Kubeconfig for user ${name} created"
 }
+
+# Use images from quay registry in order to test issues such as SRVCOM-1873
+# during upgrades. The issue is only reproducible if Knative version is identical
+# before/after upgrade but the migrator Job spec/image changes. Before upgrade,
+# the images are pulled from CI registry and after upgrade they're pulled from
+# quay.io. This way the Job spec is changes even though Knative version
+# remains same.
+function override_storage_version_migration_images {
+  local csv images name version
+  csv=${1:?Pass csv as arg[1]}
+  # Get all storage version migration images.
+  while IFS=$'\n' read -r line; do
+    images+=("$line");
+  done < <(grep storage-version-migration "$csv" | grep "image:" | awk '{ print $2 }' | awk -F"\"" '{ print $2 }')
```

Comment:
I did not include this pod/job in the checks because we haven't had problems with it, it already contains the product version, not upstream component versions (see below 1.23.0):
```
pod/kafka-controller-post-install-1.23.0--1-6vc5z
```
So, I think it's not necessary to include it. But let me know if you really want it to be added.